crates/starknet-devnet-core/src/messaging/ethereum.rs

```diff
@@ -302,9 +302,10 @@ impl EthereumMessaging {
 ///
 /// * `log` - The log to be converted.
 pub fn message_to_l2_from_log(log: Log) -> DevnetResult<MessageToL2> {
-    let parsed_log = <LogMessageToL2 as EthLogDecode>::decode_log(&log.into()).map_err(|e| {
-        Error::MessagingError(MessagingError::EthersError(format!("Log parsing failed {}", e)))
-    })?;
+    let parsed_log =
```

Comment:
what can I propose to move the extraction `l1_transaction_hash` before creating parsed_log, to avoid cloning the `Log` struct